pykern/pkcli/fmt.py

```diff
@@ -17,13 +18,19 @@ def run(path):
     _black(path)
 
 
-def diff(path):
+def diff(path_or_paths):
     """Run diff on file comparing formated vs. current file state
 
     Args:
         path (object): string or py.path to file or directory
     """
-    _black(path, "--diff", "--check", "--no-color")
+    for p in (
+        (path_or_paths,)
+        if isinstance(path_or_paths, py._path.local.LocalPath)
+        or isinstance(path_or_paths, str)
+        else path_or_paths
+    ):
+        _black(p, "--diff", "--check", "--no-color")
```

Comment:
Pass `paths` to black without expansion. This is a private interface and doesn't interact with the command line. Probably better to pass the args as a single tuple, too:

```py
_black(paths, ("--diff", "--check", "--no-color"))
```

Reply:
As far as I can tell, black doesn't accept multiple paths. If we want to only call black once we would need to use the `--include` flag with a regex based on the paths. Otherwise we need a loop somewhere. Should `_black` loop through paths?

Reply:
```sh
echo 'ok=1' > x.py
echo 'invalid python' > y.py
black --no-color x.py y.py
^[[31merror: cannot format y.py: Cannot parse: 1:8: invalid python^[[0m
^[[1mreformatted x.py^[[0m

^[[1mOh no! 💥 💔 💥^[[0m
^[[34m^[[1m1 file ^[[0m^[[1mreformatted^[[0m, ^[[31m1 file failed to reformat^[[0m.
```

Reply:
ah sorry. I think it was pksubprocess.check_call_with_signals causing the issue because it needs them separated

Reply:
81558f6e653269c07a7286f77898de84b03955fc